101-data-factory-blob-to-sql-copy/azuredeploy.json

```diff
@@ -0,0 +1,222 @@
+              }
+            }
+          },
+          {
+            "type": "Microsoft.DataFactory/datafactories/datasets",
+            "name": "[concat(parameters('dataFactoryName'), '/', variables('blobInputDatasetName'))]",
+            "dependsOn": [
+              "[concat('Microsoft.DataFactory/dataFactories/', parameters('dataFactoryName'))]",
+              "[concat('Microsoft.DataFactory/dataFactories/', parameters('dataFactoryName'), '/linkedServices/', variables('azureStorageLinkedServiceName'))]"
+            ],
+            "apiVersion": "[variables('apiVersion')]",
+            "properties": {
+              "type": "AzureBlob",
+              "linkedServiceName": "[variables('azureStorageLinkedServiceName')]",
+              "structure": [
+                {
+                  "name": "Column0",
+                  "type": "String"
+                },
+                {
+                  "name": "Column1",
+                  "type": "String"
+                }
+              ],
+              "typeProperties": {
+                "folderPath": "[concat(parameters('sourceBlobContainer'), '/')]",
+                "fileName":  "[parameters('sourceBlobName')]",
+                "format": {
+                  "type": "TextFormat",
+                  "columnDelimiter": ","
+                }
+              },
+              "availability": {
+                "frequency": "Day",
+                "interval": 1
+              },
+              "external": true
+            }
+          },
+          {
+            "type": "Microsoft.DataFactory/datafactories/datasets",
+            "name": "[concat(parameters('dataFactoryName'), '/', variables('sqlOutputDatasetName'))]",
+            "dependsOn": [
+              "[concat('Microsoft.DataFactory/dataFactories/', parameters('dataFactoryName'))]",
+              "[concat('Microsoft.DataFactory/dataFactories/', parameters('dataFactoryName'), '/linkedServices/', variables('azureSqlLinkedServiceName'))]"
+            ],
+            "apiVersion": "[variables('apiVersion')]",
+            "properties": {
+              "type": "AzureSqlTable",
+              "linkedServiceName": "[variables('azureSqlLinkedServiceName')]",
+              "structure": [
+                {
+                  "name": "FirstName",
+                  "type": "String"
+                },
+                {
+                  "name": "LastName",
+                  "type": "String"
+                }
+              ],
+              "typeProperties": {
+                "tableName": "[parameters('targetSQLTable')]"
+              },
+              "availability": {
+                "frequency": "Day",
+                "interval": 1
+              }
+            }
+          },
+          {
+            "type": "Microsoft.DataFactory/datafactories/datapipelines",
```

Comment:
since these are child resources, remove the extra name segments to improve readability.